backend/src/endpoints/v1/users.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from uuid import UUID
+
+import sqlalchemy.exc
+from fastapi import APIRouter, Depends, status, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.services.users import UserService
+
+from src.core.schemas.user.response import (
+    UserCreateResponseSchema,
+    PaginationResponseSchema,
+    UserUpdateResponseSchema,
+)
+from src.core.schemas import MessageResponseSchema
+from src.core.deps import with_async_session
+from src.core.schemas import ControllerConfig
+
+router = APIRouter(prefix="/users", tags=["Users"])
+
+
+def get_user_service(session: AsyncSession = Depends(with_async_session)):
+    return UserService(session=session)
+
+
+def user_service_pagination(
+    session: AsyncSession = Depends(with_async_session), config: ControllerConfig = Depends(ControllerConfig)
+):
+    return UserService(session=session, config=config)
+
+
+@router.get("", summary="get users", response_model=PaginationResponseSchema)
+async def users(user_service: UserService = Depends(user_service_pagination)):
+    users_res_test = await user_service.list_users()
+
+    return users_res_test
+
+
+@router.get("/{id}")
+async def get_user_by_id(user_id: UUID, service: UserService = Depends(get_user_service)):
+    user = await service.retrieve_user(user_id)
+    if user is None:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    return {"data": user}
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED, response_model=UserCreateResponseSchema | MessageResponseSchema)
+async def create_user(user: UserCreateResponseSchema, service: UserService = Depends(get_user_service)):
+    try:
+        user_obj = await service.create_user(user)
+        return {"data": user_obj}
+    except sqlalchemy.exc.IntegrityError as e:
+        logging.error(e)
+        return {"message": "Failed to create user"}
```

Comment:
response_model=UserCreateResponseSchema | MessageResponseSchema
У нас вот эта строчка нужна для того чтобы результат который ты возвращаешь из endpointa должно быть или UserCreateResponseSchema или MessageResponseSchema, ты можешь просто вернуть user_obj и у тебя эти схемы обработают твой объект из бд

```suggestion
@router.post("/", status_code=status.HTTP_201_CREATED, response_model=UserCreateResponseSchema | MessageResponseSchema)
async def create_user(user: UserCreateResponseSchema, service: UserService = Depends(get_user_service)):
        return await service.create_user(user)
```